docs/configuration.md

```diff
@@ -183,7 +183,7 @@ Most of the following properties are designed for performance tuning. Adjusting
 * `enable_envvars_config`: Enable configuring TorchServe through environment variables. When this option is set to "true", all the static configurations of TorchServe can come through environment variables as well. Default: false
 * `number_of_netty_threads`: number frontend netty thread. Default: number of logical processors available to the JVM.
 * `netty_client_threads`: number of backend netty thread. Default: number of logical processors available to the JVM.
-* `default_workers_per_model`: number of workers to create for each model that loaded at startup time. Default: available GPUs in system or number of logical processors available to the JVM.
+* `default_workers_per_init_model`: number of workers to create for each model that loaded at startup time. Default: available GPUs in system or number of logical processors available to the JVM.
```

Comment:
Why are we modifying the original configuration? Can't we simply change the snapshot to follow what exists in main configuration?

Reply:
@vdantu : The actual configuration parameter documented is `default_workers_per_model`. However, the config manager internally uses both `default_workers_per_model` and `NUM_WORKERS`. Further, the snapshots generated is a combination of config properties and state of all registered models. The config manager dumps `NUM_WORKERS` param in the snapshot instead of `default_workers_per_model`.

Thus it makes sense to use only the documented parameter.

We are also renaming the config parameter as it is only applicable for the models which are loaded/initialized at the time of starting `torchserve`. For all other models registered through API, the user needs to provide the number of workers.

Reply:
I was thinking from customers view point. If a customer was using previously released version of TorchServe and they are moving to newer version of TorchServe, would changing this option break their experience or not? Variables used internally can be modified while maintaining backwards compatibility would be my thought here. Changing the customer experience here seems odd. Can't we keep this variable as it is and change the `NUM_WORKERS` to `current_num_workers` or to some **other better name**? Also, `init` in `default_workers_per_init_model` seems redundant. I am open for other views here, but it seems like we are changing UX for something that can be fully kept internal to codebase. Correct me if I am wrong.

Reply:
The only idea to introduce the `init` keyword here is to indicate that it applies to the startup/initialization models and not to every model.

Just `default_workers_per_model` sounds like it will apply to every model if no worker is provided.

I agree this will impact the UX where the parameter is already being set in config.properties.